### PR TITLE
add SetupTools to auto add Planes and XRRemote-Debug layers on editor…

### PR DESCRIPTION
**Pull request by Razieleron**
SetupTools.cs runs a script on editor load to add Plane and XRRemote-Debug to the layers list. Additionally, there is a menu button called 'Tools' with an option to 'Setup Layers' which does the same thing, but you shouldn't have to click it since the method is run on editor startup. We can add additional functionality to this later if we want.

There is some error handling in the creation of the layers to make sure we're not overwriting an already labeled layer (we just target the next unlabeled layer) and an error log if for some reason there are no available layers whatsoever.

### CODE REVIEW

**Review by CarlSustrich — COMMENTED** [review]
Looks solid! 
Discovered, if the 'culling mask' feature on the camera component of the ARCamera object is set to 'mixed', the created layers will not automatically be added. I'd imagine our optimal solution would involve automatically enabling these layers in the ARCamera upon creation. Seems too fiddly for end user to need to go enable them themselves?
[This](https://docs.unity3d.com/ScriptReference/Camera-cullingMask.html) method could be pertinent. Would need to make sure we're only targeting the AR Camera to add the layers to, not a standard UI camera or similar.

**Comment by gblikas** [comment]
> Discovered, if the 'culling mask' feature on the camera component of the ARCamera object is set to 'mixed', the created layers will not automatically be added. I'd imagine our optimal solution would involve automatically enabling these layers in the ARCamera upon creation. Seems too fiddly for end user to need to go enable them themselves?

@Razieleron yes, please make sure that the layers you added are being rendered after adding.

**Comment by Razieleron** [comment]
@gblikas wrapped up that culling mask issue.

**Review by gblikas — CHANGES_REQUESTED** [review]
@Razieleron took a look at the code and it looks like a couple changes are needed. 
1. please make sure that all files have our License agreement at the top. 
2. make sure any sources that were used to create the code are cited. 
3. the SetupTools functionality shouldn't be split between `XRRemote/Layer Utilities/Create Layers` and `XRRemote/Layer Utilities/Setup Camera Culling Mask`. if you're setting up the project, it should make sure that layers _and_ the culling mask is setup.

**Comment by Razieleron** [comment]
The code now successfully targets all AR cameras in the hierarchy and sets their culling mask. 

Upon thinking about it, Carl and I figured that it could get obnoxious for the end user if the Planes and XRRemote-Debug layers were set to checked on all AR cameras on every script recompilation/editor refresh and so we localized that functionality to the XRRemote -> Layer Utilities -> Create Layers and Setup Camera Culling Mask button in the menu.

If you would like it to be set up differently, please let me know.

**Review by CarlSustrich — DISMISSED** [review]
New changes - adds layers to culling mask directly after creating them, and this only occurs on menu item select by user, not automatically on load. Looks good to me 🤘

File: Assets/Editor/SetupTools.cs [hunk 1]
```diff
@@ -0,0 +1,124 @@
+using System;
+using System.Runtime.CompilerServices;
```
Comment (gblikas) [inline Assets/Editor/SetupTools.cs:2]: using statement needs to go below license statement

**Review by gblikas — CHANGES_REQUESTED** [review]

File: Assets/Editor/SetupTools.cs (hunk 1 above)
Comment (Razieleron) [reply Assets/Editor/SetupTools.cs]: I've moved them below the license statement-

**Review by gblikas — APPROVED** [review]

**Pull request merged**